### ISSUE TITLE
Download links should include link download instructions

### ISSUE THREAD

**Issue by Rick-Anderson** [aspnet/Docs#4009]
URL Rewriting Middleware in ASP.NET Core | Microsoft Docs 
[08-11-2017] The article states 'View or download sample code', but there is no way to download the sample code from Github!

Append the following to download links.
`[how to download](xref:tutorials/index#how-to-download-a-sample)`

**Comment by guardrex** [comment]
I worked on this on the .NET Core side:

https://github.com/dotnet/docs/issues/1644
https://github.com/dotnet/docs/pull/1832

**Comment by Rick-Anderson** [comment]
IIRC, the .NET download instructions were very complex - talking about forking and other git concepts. The folks who struggle with downloads need only to be told to nav to the top of the repo and download the entire repo. Unzip then navigate locally to the sample. Because the .NET version was so complex, I wrote a simpler version. I can't find it now. Maybe it got deleted.

**Comment by guardrex** [comment]
Yes ... IIRC it was decided to let the content on that topic explain it all out and just link to it ...

> For download instructions, see \[Samples and Tutorials](../../samples-and-tutorials/index.md#viewing-and-downloading-samples).

It's at ...

https://docs.microsoft.com/en-us/dotnet/samples-and-tutorials/#viewing-and-downloading-samples

**Comment by Rick-Anderson** [comment]
Yep, way to complex. It doesn't tell you to navigate from the link provided to the repo. Then it has all the fork and clone nonsense that you don't need to download a sample. I rewrote the instructions in 10% of the words and it was super easy to follow.

**Comment by guardrex** [comment]
I hope u find that ... at the time, it wasn't known that there was another way to *only* get the contents of the sample folder. That was the big prob IIRC.

**Comment by tdykstra** [comment]
```
[how to download](xref:tutorials/index#how-to-download-a-sample)
```

**Comment by guardrex** [comment]
> Use the URL in the sample link to help you navigate to the sample directory.

Lovely. :grinning:

**Comment by guardrex** [comment]
Had the same mention over in the other text ...

> The relative path to the sample's location appears in your browser's address bar when you follow the link to the sample.

... and in fairness to the complexity, the key is that it's providing options for those methods ...

> Download the repository that contains the sample code by performing **one of the following procedures**: (emphasis added)

... and one of those listed options is to just download the ZIP.

I favor showing the options. It cross-teaches a little :sparkles: GH magic :sparkles: (with links) ... prob not a bad thing. If a GH newbie learns how to fork and clone, they're a stone's throw away from taking some .NET docs issues more complicated than patching a misspelled word. :wink:

**Comment by ardalis** [comment]
If we could nudge GitHub into supporting Zipping of arbitrary folders for download (which I believe some other competing sites offer), this would be a non-issue.

**Comment by Rick-Anderson** [comment]
@BillWagner - GitHub ask for you.
> If we could nudge GitHub into supporting Zipping of arbitrary folders for download (which I believe some other competing sites offer), this would be a non-issue.

TLDR; version
Customers don't know how to download a sample when we give them the download link in an article.  The .NET help on download is insanely complex, mixing up forks and other complexities new folks don't need. I wrote [How to download a sample](https://docs.microsoft.com/en-us/aspnet/core/tutorials/#how-to-download-a-sample) - and no complaints on that one.
